src/hv_exc.c

```diff
@@ -424,7 +441,7 @@ void hv_exc_fiq(struct exc_info *ctx)
         tick = true;
     }
 
-    if (mrs(TPIDR_EL2) != 0 && !(mrs(ISR_EL1) & 0x40) && hv_want_cpu == -1) {
```

Comment:
Then this needs to check for `hv_want_cpu == -1 && hv_pinned_cpu != mrs(TPIDR_EL2)`, and the below check for `mrs(TPIDR_EL2) == 0` also needs to change to `== hv_pinned_cpu` only if it is not -1.